mpas_analysis/ocean/streamfunction_moc.py

```diff
@@ -345,34 +364,48 @@ def _compute_moc_climo_postprocess(self):  # {{{
 
         config = self.config
 
-        dvEdge, areaCell, refBottomDepth, latCell, nVertLevels, \
-            refTopDepth, refLayerThickness = self._load_mesh()
-
         self.regionNames = config.getExpression(self.sectionName,
                                                 'regionNames')
 
         # Load basin region related variables and save them to dictionary
+        mpasMeshName = config.get('input', 'mpasMeshName')
+        regionMaskDirectory = config.get('regions', 'regionMaskDirectory')
+
+        regionMaskFile = '{}/{}_MOCBasinsAndTransectMasks.nc'.format(
+                regionMaskDirectory, mpasMeshName)
+
+        if not os.path.exists(regionMaskFile):
+            raise IOError('Regional masking file {} for MOC calculation '
+                          'does not exist'.format(regionMaskFile))
         # NB: The following will need to change with new regional mapping files
-        regionMaskFiles = config.get(self.sectionName, 'regionMaskFiles')
-        if not os.path.exists(regionMaskFiles):
+        regionMaskFile = config.get(self.sectionName, 'regionMaskFile')
+        if not os.path.exists(regionMaskFile):
             raise IOError('Regional masking file for MOC calculation '
                           'does not exist')
-        iRegion = 0
         self.dictRegion = {}
+        dsMask = xr.open_dataset(regionMaskFile)
         for region in self.regionNames:
             print '\n  Reading region and transect mask for ' \
                 '{}...'.format(region)
-            ncFileRegional = netCDF4.Dataset(regionMaskFiles, mode='r')
-            maxEdgesInTransect = \
-                ncFileRegional.dimensions['maxEdgesInTransect'].size
-            transectEdgeMaskSigns = \
-                ncFileRegional.variables['transectEdgeMaskSigns'][:, iRegion]
-            transectEdgeGlobalIDs = \
-                ncFileRegional.variables['transectEdgeGlobalIDs'][iRegion, :]
-            regionCellMask = \
-                ncFileRegional.variables['regionCellMasks'][:, iRegion]
-            ncFileRegional.close()
-            iRegion += 1
+
+            found = False
+            iRegion = 0
+            for mocRegion in dsMask.regionNames.values:
```

Comment:
The previous code for determining the region index `iRegion` was incorrect.  It just used the index into the list of regions to be plotted in the config file.  I have fixed that here by looping over regions *in the file* and finding the one matching the requested region (with `_MOC` added to the region name in the file).  Let me know if this makes sense to you.

We probably could have subset the data set using `dsMask.sel(nRegion='{}_MOC'.format(region))` but I wasn't sure that would work or if it would be clear to developers...